docs/proposals/scheduling/20240201-enbale-reservation-preempt.md

```diff
@@ -0,0 +1,171 @@
+- `scheduling.koordinator.sh/reservation-priority` is filled by user, describe reservation 's priority, only higher
+  priority reservations can preempt lower priority reservations.
+
+#### Preemption
+The reservation preemption still follows the existing Filter/PostFilter procedure and can be combined with job-level
+preemption mechanism.
+
+The preemption strategy of reservation is as followed:
+1. Only preemptionPolicy=PreemptLowerPriority reservation can trigger preempt, and only lower priority reservations can be preempted.
+
+2. Reservations can only preempt reservations.
+
+3. If only part of Reservation can be assigned successfully in preemption dry-run process, the preemption will not
+   really happen.
+
+#### Response To Preempted
+Once a reservation is chosen to be evicted, it will follow the scheduler implemented eviction mechanism, support soft-eviction
+or delete according to the implementation. The only difference is that we need to check the evicted is pod or reservation.
+
+#### Extension Point
+
+##### Over All
+Generally we will extend the job-oversold plugin, and modify other plugins to support reservation preemption.
+The new\delta parts are:
+1. Enable Reservation pod to preempt.
+2. Enable Reservation pod to be preempted.
+3. patch evict label/annotation to Reservation.
+4. Register Reservation eventHandler for job-oversold plugin.
+
+##### PreFilter
+If pod is reservation pod, as reservation is not associated to quota yet, so it is no need to do quota check for reservation.
+
+##### PostFilter
+We will maintain the existing implementation process, but with the following differences.
+
+1. Skip the quota check.
+2. Reservation pods only preempt reservation pods.
+
+### API
+#### Reservation
+We introduce some labels to describe reservation behaviour.
+- `pod-group.scheduling.sigs.k8s.io` is filled by user, describe reservation should be scheduled in batch.
+- `scheduling.koordinator.sh/soft-eviction` is filled by scheduler, indicate the reservation is preempted.
+- `spec.preemptionPolicy` is filled by user, describe whether a reservation can trigger preemption.
+- `scheduling.koordinator.sh/reservation-priority` is filled by user, describe reservation 's priority, only higher
+  priority reservations can preempt lower priority reservations.
+
+If you want to declare reservation belongs to the same batch, please use as follows:
+```yaml
+labels:
+  pod-group.scheduling.sigs.k8s.io: "reservation-batch-1"
+```
+
+if you want to declare reservation can trigger preemption, please use as follows:
+```yaml
+spec:
+  preemptionPolicy: PreemptLowerPriority
+```
+
+if you want to check reservation preempted message, please focus on label:
+```yaml
+labels:
+  scheduling.koordinator.sh/soft-eviction: "{preemptReservation:reservation2...}"
+```
+
+if you want to set reservation priority, please use as follows:
+```yaml
+labels:
+  scheduling.koordinator.sh/reservation-priority: "9900"
+```
```

Comment:
These contents have been introduced in previous chapters. It is recommended to add some examples to show users how to set fields so that the Reservation can preempt others, or be preempted by others.

Reply:
done